src/tests/test_core_python.py

```diff
@@ -1,70 +1,99 @@
+from functools import cache
 from pathlib import Path
+from typing import Any
 
 import pytest
+import yaml
+from yaml import CLoader as Loader
 
 from pyodide_build.common import UNVENDORED_STDLIB_MODULES
 
 
-def test_cpython_core(python_test, selenium, request):
-
-    name, error_flags = python_test
-
+def filter_info(info: dict[str, Any], browser: str) -> dict[str, Any]:
+    info = dict(info)
     # keep only flags related to the current browser
     flags_to_remove = ["firefox", "chrome", "node"]
-    flags_to_remove.remove(selenium.browser)
-    for flag in flags_to_remove:
-        if "crash-" + flag in error_flags:
-            error_flags.remove("crash-" + flag)
+    flags_to_remove.remove(browser)
+    for browser in flags_to_remove:
+        for key in list(info.keys()):
+            if key.endswith(browser):
+                del info[key]
+    return info
 
-    if any(flag.startswith("segfault") for flag in error_flags):
-        pytest.skip('known segfault: "{}"'.format(" ".join(error_flags)))
 
-    if error_flags:
+def possibly_skip_test(request, info: dict[str, Any]) -> dict[str, Any]:
+    for reason in (
+        reason for [flag, reason] in info.items() if flag.startswith("segfault")
+    ):
+        pytest.skip(f"known segfault: {reason}")
+
+    for reason in (
+        reason for [flag, reason] in info.items() if flag.startswith("xfail")
+    ):
         if request.config.option.run_xfail:
             request.applymarker(
                 pytest.mark.xfail(
                     run=False,
-                    reason='known failure: "{}"'.format(" ".join(error_flags)),
+                    reason=f"known failure: {reason}",
                 )
             )
         else:
-            pytest.xfail('known failure: "{}"'.format(" ".join(error_flags)))
+            pytest.xfail(f"known failure: {reason}")
+    return info
+
+
+def test_cpython_core(main_test, selenium, request):
+    [name, info] = main_test
+    info = filter_info(info, selenium.browser)
+    possibly_skip_test(request, info)
+
+    ignore_tests = info.get("skip", [])
+    if not isinstance(ignore_tests, list):
+        raise Exception("Invalid python_tests.yaml entry: 'skip' should be a list")
 
     selenium.load_package(list(UNVENDORED_STDLIB_MODULES))
     try:
         selenium.run(
+            f"""
+import platform
+from test import libregrtest
```

Comment:
maybe keep the original indentation and add a textwrap.dedent around it, if necessary, for better readability?